registration-processor/core-processor/registration-processor-reprocessor-stage/src/main/java/io/mosip/registration/processor/reprocessor/service/WorkflowActionService.java

```diff
@@ -220,17 +231,26 @@ private void processResumeFromBeginningAndRemoveHotlistedTag(List<String> workfl
 				try {
 
 				removeHotlistedTag(rid);
-				InternalRegistrationStatusDto registrationStatusDto = getAndUpdateRegistrationStatus(workflowActionCode,
-						rid, RegistrationStatusCode.PROCESSING);
-				if (RegistrationStatusCode.REPROCESS_FAILED.name().equals(registrationStatusDto.getStatusCode())) {
-					description.setMessage(PlatformErrorMessages.RPR_WAS_REPROCESS_FAILED.getMessage());
-				} else {
-				sendPacketEventforResumeBeginning(registrationStatusDto);
-				description.setMessage(
-							String.format(PlatformSuccessMessages.RPR_WORKFLOW_ACTION_SERVICE_SUCCESS.getMessage(),
-									workflowActionCode.name()));
+				InternalRegistrationStatusDto registrationStatusDto = getRegistrationStatus(rid);
+				if (registrationStatusDto.getStatusCode().equalsIgnoreCase(RegistrationStatusCode.PAUSED.name())) {
+					registrationStatusDto = updateRegistrationStatus(registrationStatusDto,
+							RegistrationStatusCode.PROCESSING,
+							workflowActionCode);
+					if (RegistrationStatusCode.REPROCESS_FAILED.name().equals(registrationStatusDto.getStatusCode())) {

```

Comment:
Is this If condition required, if updateRegistrationStatus method din't throw any exception, then this should be always true rt ?

Reply:
Same query for all below cases